### PR TITLE
chore: introduce update-snapshots=changed

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/test-cli-js.md [hunk 1]
```diff
@@ -76,33 +76,40 @@ Here are the most common options available in the command line.
...
-| `--forbid-only` | Whether to disallow `test.only`. Useful on CI.|
-| `--global-timeout <number>` | Total timeout for the whole test run in milliseconds. By default, there is no global timeout. Learn more about [various timeouts](./test-timeouts.md).|
-| `-g <grep>` or `--grep <grep>` | Only run tests matching this regular expression. For example, this will run `'should add to cart'` when passed `-g "add to cart"`.  The regular expression will be tested against the string that consists of the project name, test file name, `test.describe` titles if any, test title and all test tags, separated by spaces, e.g. `chromium my-test.spec.ts my-suite my-test @smoke`. The filter does not apply to the tests from dependency projects, i.e. Playwright will still run all tests from [project dependencies](./test-projects.md#dependencies). |
-| `--grep-invert <grep>` | Only run tests **not** matching this regular expression. The opposite of `--grep`. The filter does not apply to the tests from dependency projects, i.e. Playwright will still run all tests from [project dependencies](./test-projects.md#dependencies).|
-| `--headed` | Run tests in headed browsers. Useful for debugging. |
```
Comment (yury-s) [inline docs/src/test-cli-js.md:89]: why did anything but snapshots change here? revert?

**Review by yury-s — APPROVED** [review]

File: docs/src/test-cli-js.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/src/test-cli-js.md:89]: I aligned this doc with the --help in cli

File: packages/playwright/src/matchers/toMatchSnapshot.ts [hunk 2]
```diff
@@ -317,17 +317,33 @@ export function toMatchSnapshot(
...
 
   const expected = fs.readFileSync(helper.expectedPath);
-  const result = helper.comparator(received, expected, helper.options);
-  if (!result)
-    return helper.handleMatching();
 
   if (helper.updateSnapshots === 'all') {
+    if (!compareBuffersOrStrings(received, expected)) {
+      /* eslint-disable no-console */
+      console.log(helper.expectedPath + ' is the same, not writing actual.');
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply packages/playwright/src/matchers/toMatchSnapshot.ts:324]: The user of `all` might be confused as to why some of the files are not touched.

[the remark this answers is not in the document]
Comment (pavelfeldman) [reply packages/playwright/src/matchers/toMatchSnapshot.ts:324]: I'll remove it for now for consistency though!

**Pull request merged**